crates/primitives/starknet/src/transaction/types.rs

```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
         }
     }
 }
+
+/// Declare transaction.
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    Eq,
+    scale_codec::Encode,
+    scale_codec::Decode,
+    scale_info::TypeInfo,
+    scale_codec::MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct DeclareTransaction {
```

Comment:
do we support `DeclareV2` directly ? Or is that V0V1 ?

Reply:
It's V1, v0 is deprecated and we don't support cairo 1 yet